pkg/core/storage/memcached_store.go

```diff
@@ -100,11 +100,12 @@ func (s *MemCachedStore) Seek(rng SeekRange, f func(k, v []byte)) {
 func (s *MemCachedStore) SeekAsync(ctx context.Context, rng SeekRange, cutPrefix bool) chan KeyValue {
 	res := make(chan KeyValue)
 	go func() {
-		s.seek(ctx, rng, cutPrefix, func(k, v []byte) {
+		s.seek(ctx, rng, cutPrefix, func(k, v []byte) bool {
 			res <- KeyValue{
 				Key:   k,
 				Value: v,
 			}
+			return true // always continue, we have context for early stop.
```

Comment:
We could check `ctx.Done()` here, but it seems to be unnecessary.

Reply:
Yes, `ctx.Done` is being checked inside of `(s *MemCachedStore) seek`, so  I think it's enough.